verilog/analysis/verilog_filelist.h

```diff
@@ -32,20 +32,39 @@ struct TextMacroDefinition {
 };
 
 // File list for compiling a System Verilog project.
+// TODO: ideally, all the strings would be string_views, but we need to make
+//       sure to have the relevant backing store live.
+// TODO: are there cases in which files and incdirs are interleaved so that
+//       the first files don't see all the incdirs yet, but after more incdirs
+//       are added, all of them are relevant ? If so, this would require some
+//       restructering.
+// TODO: document if files are relative to tool invocation or relative to
+//       file_list_path (the latter makes more sense, but I think currently that
+//       is underspecified).
+// TODO: Alongside previous: also introduce file_list_root field ?
```

Comment:
Would that mean that paths in `file_paths` should exist under `file_list_root`?

Reply:
Also, I don't get the first TODO.
Isn't `file_list_path` a file that contains the arguments (files, defines and incdirs)? why should files passed through the command line be relative to it?

Reply:
That TODO is mostly for me to figure out as I don't really know what happens if file_list_root is not specified.

The list of files in the file-list either are absolute `/path/to/file/...` or relative `relative/to/somedir`, and I suspect file_list_root is what provides the prefix `${file_list_root}/relative/to/some/dir`.
However, in absence of `file_list_root` specifically defined, is that just `$(dirname file_list_path)` ? Probably yes, but since I don't know yet I wrote it as TODO for me to figure out.

From the command line, the situation is more clear: any files are provided relative to the current working directory.